File: backend/Core/Services/TarefaService.cs
using backend.Core.Interfaces;
using backend.Core.Interfaces.Repositories;
using backend.Core.Models;

namespace backend.Core.Services
{
    public class TarefaService : ITarefaService
    {
        private readonly ITarefaRepository _tarefaRepository;

        public TarefaService(ITarefaRepository tarefaRepository)
        {
            _tarefaRepository = tarefaRepository;
        }

        public async Task<List<Tarefa>> ListarTarefas()
        {
            return await _tarefaRepository.ListarTarefas();
        }

        public async Task<Tarefa> CadastrarTarefa(Tarefa tarefa)
        {
            return await _tarefaRepository.CadastrarTarefa(tarefa);
        }

        public async Task<Tarefa> BuscarTarefaPorId(int id)
        {
            return await _tarefaRepository.BuscarTarefaPorId(id);
        }

        public async Task<Tarefa> EditarTarefa(Tarefa tarefa)
        {
            return await _tarefaRepository.EditarTarefa(tarefa);
        }

        public async Task<bool> ExcluirTarefa(int id)
        {
            return await _tarefaRepository.ExcluirTarefa(id);
        }
    }
}

File: frontend/src/app/services/tarefa.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { tarefa } from '../interfaces/tarefa';
import { HttpClient } from '@angular/common/http';
@Injectable({
    providedIn: 'root'
})

export class TarefaService {

    private apiUrl = 'http://localhost:5248/api/Tarefa';

    constructor(private http: HttpClient) { }

    cadastrarTarefa(tarefa: tarefa) {
        return this.http.post(`${this.apiUrl + "/cadastrarTarefa"}`, tarefa);

    }

    editarTarefa(tarefa: tarefa) {
        return this.http.put(`${this.apiUrl}/editarTarefa/${tarefa.id}`, tarefa);
    }

    excluirTarefa(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/excluirTarefa/${id}`);
    }

    buscarTarefaPorId(id: number): Observable<tarefa> {
        return this.http.get<tarefa>(`${this.apiUrl}/buscarTarefaPorId/${id}`);
    }

    listarTarefa(): Observable<tarefa[]> {
        return this.http.get<tarefa[]>(this.apiUrl);
    }
}
